cfme/tests/infrastructure/test_host.py

```diff
@@ -312,3 +312,41 @@ def _finalize():
             actual_state, power_state
         ),
     )
+
+
+@test_requirements.infra_hosts
+def test_infrastructure_hosts_refresh_multi(appliance, setup_provider, provider):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: low
+        initialEstimate: 1/6h
+        testSteps:
+            1. Navigate to the Compute > Infrastructure > Providers view.
+            2. Click on a provider guadicon, and then the hosts link along the top row of the view.
+            3. Select all hosts (need at least 2 hosts) by checking the box in upper left of
+               quadicons.
+            4. Click "Refresh Relationships and Power States" under the Configuration
+               dropdowm, and then click "OK" when prompted.
+        expectedResults:
+            1. Providers view is displayed.
+            2. Hosts view is displayed.
+            3.
+            4. "Refresh initiated for X Hosts from the CFME Database" is displayed in green
+               banner where "X" is the number of selected hosts. Properties for each host are
+               refreshed.
+    """
+    hosts_view = navigate_to(provider.collections.hosts, "All")
+    num_hosts = len(hosts_view.entities.get_all())
+    if num_hosts < 2:
+        pytest.skip('not enough hosts to run test')
+    for h in hosts_view.entities.get_all():
+        h.check()
+    hosts_view.toolbar.configuration.item_select('Refresh Relationships and Power States',
+                                                 handle_alert=True)
+    hosts_view.flash.assert_success_message(
+        f'Refresh initiated for {num_hosts} Hosts from the CFME Database'
+    )
+    assert wait_for(provider.is_refreshed, func_kwargs={"refresh_delta": 120}, num_sec=300,
```

Comment:
you don't need assert here since it will raise Timedout exception

Reply:
@izapolsk is it necessary to wrap? I agree with you on assert, but if we do just `wait_for(provider.is_refreshed...)` and it fails, the test fails as well.

Reply:
So we are not expecting each test to contain an assert? I am not arguing this point, just getting clarification. I am accustomed to each test having one, and only one assert (sometimes it makes sense to have more than one, but only in special cases). If the team is OK with tests passing with no asserts if they don't explicitly fail, I will comply.

Reply:
What is specifically wrong with the assert? Are we saying that the assert may cause/allow an incorrect test result, or do we just not like the explicit assert here. I'd definately want to fix anything that cases bad test results, but I don't see anything wrong with an explicit assert. It is what I am checking. If the wait_for times out, the test will fail. Otherwise the view is displayed and the test passes (exactly what I want). 
I have not tested the failure case (this test passes on the current builds), but the positive case seems to work. Do I need to cause the failure to make sure it would cause a fail result?

Reply:
@lina-nikiforova, there is a difference between Failed test and Error(Broken) test. I'm certain you understand the difference. From my point, yes, we need to wrap wait_for calls used to check test codition into try/expect and raise pytest.fail() in order to mark test as Failed not Broken.
btw, I don't say that we need to wrap all wait_for calls. I mean only wait_for calls in test which check some test condition.

@prichard77, let me explain what will happen there.
1. if condition is True in wait_for, it will pass True to your assert and test will be passed.

2. If condition in False in wait_for, wait_for will call that method until is reaches timeout value.
wait_for will raise TimedOutException in such case. So, your assert will never get False.
so, it's redundant. Btw, your test will be marked as Error(Broken) instead of Failed in such case.

So, what I think you need to do
```
try:
  wait_for(provider.is_refreshed, func_kwargs={"refresh_delta": 120}, num_sec=300, delay=30)
except TimedOutError:
   pytest.fail("Blahblah wasn't refreshed within given time")
```

Reply:
@izapolsk thanks for the explanation! I agree with this point.

Reply:
I agree with the need to use try/except to better represent timeout as a failure and not error. I still question what is wrong with an assert? What about:
```
try:
  assert wait_for(provider.is_refreshed, func_kwargs={"refresh_delta": 120}, num_sec=300, delay=30)
except TimedOutError:
   pytest.fail("Blahblah wasn't refreshed within given time")

Reply:
@prichard77, there is nothing wrong with assert. it will just never be used in this particular case. because False will never be passed to it. so, it doesn't matter if it present or absent.